File: Introduction to Computing/Algorithms Needed/DIA_bsearch.py
n = [1,2,3,4,5,6,7,8,9,10]
s = 6
l = 0
r = len(n)-1
found = "NO"

while found != "YES" and l <= r:
	m = int((l+r)/2);
	if n[m] == s:
		found = "YES"
	elif n[m] < s: l = m + 1
	else: r = m-1
if found == "YES":
	print(s, "found in index", m)
else:
	print(s, "not found")

File: Introduction to Computing/Algorithms Needed/DIA_bsearch.cpp
/*
Dia, Christian A.
CSDC100 ZC11
9. 24. 19
Algorithm Implementation:

*/


#include<iostream>
#include<algorithm>
using namespace std;

int main(){
	int num, i; 
	cout << "Enter size: "; cin >> num;
	int n[num] = {};
	cout << "Enter values: ";
	for(i = 0; i < num; ++i) cin >> n[i];
	int l = 0, r = num, m, s = 6;
	cout << "Enter value to search: "; cin >> s;
	bool found = false;
	sort(n, n+num);
	while(!found && r >= l){
		m = (r+l)/2;
		if(n[m] == s) found = true;
		else if(n[m] > s) r = m-1;
		else l = m + 1;
	}
	found ? cout << s << " found at index " << m : cout << s << " Not found!";
	cout << endl; 
	return 0;
}
